nbdsrv.h

```diff
@@ -59,6 +59,7 @@ typedef struct _client {
 	GArray *export;    /**< array of FILE_INFO of exported files;
 			       array size is always 1 unless we're
 			       doing the multiple file option */
+	pthread_mutex_t export_lock;
```

Comment:
this needs to be a `pthread_rwlock_t` rather than a `pthread_mutex_t`, presumably?